### ISSUE TITLE
LD Flag `-s` Obselete

### ISSUE THREAD

**Issue by btnwtn** [google/guetzli#41]
Building guetzli via `make` on Mac OS X 10.11.3 results in this error for me:

```
ld: warning: option -s is obsolete and being ignored
ld: internal error: atom not found in symbolIndex(__ZN7guetzli21ButteraugliComparatorD2Ev) for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[1]: *** [bin/Release/guetzli] Error 1
make: *** [guetzli] Error 2
```

To fix this error I edited `guetzli.make` and made these changes:

```diff
- ALL_LDFLAGS += $(LDFLAGS) -s
+ ALL_LDFLAGS += $(LDFLAGS) #-s
```

This successfully linked the build and generated the guetzli executable in `bin/Release`.
I don't know anything about make, but if someone could point me in the proper direction for detecting OS X in a make file I would be willing to open a PR to fix this issue.

Cheers!

**Comment by fkistner** [comment]
That's weird.
I tried making guetzli available in Homebrew, see Homebrew/homebrew-core#11190.
The homebrew build bot did not encounter this problem on [El Capitan](https://bot.brew.sh/job/Homebrew%20Core/18580/version=el_capitan/testReport/brew-test-bot/el_capitan/install_guetzli/), neither did my local test on Sierra. It did however on [Yosemite](https://bot.brew.sh/job/Homebrew%20Core/18580/version=yosemite/testReport/junit/brew-test-bot/yosemite/install_guetzli/).

**Comment by robryk** [comment]
This seems to be something related to premake/premake-core#133, premake/premake-core#270 and http://industriousone.com/topic/how-remove-flags-ldflags

I don't really understand the problem right now; I'll work it around by actually causing debugging symbols to be emitted (this doesn't impact performance).

@fkistner @btnwtn Are you both using Clang?

**Comment by fkistner** [comment]
Yes, Homebrew uses clang by default, but uses a custom cc script to inject additional command line options that tend to be necessary.
I don't have a Yosemite system to reproduce, but here are the actual command invocations for Sierra (for which it works fine): https://gist.github.com/fkistner/effbd64115a382df9bfe7e3f500b1046

**Comment by robryk** [comment]
This should now be effectively worked around. If it still manifests, please comment here. I'm keeping the issue open until we figure out what's actually happening.
